packages/webview_flutter/lib/src/webview_method_channel.dart

```diff
@@ -32,6 +32,42 @@ class MethodChannelWebViewPlatform implements WebViewPlatform {
     });
   }
 
+  @override
+  Future<void> updateSettings(WebSettings settings) {
+    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+    // https://github.com/flutter/flutter/issues/26431
+    // ignore: strong_mode_implicit_dynamic_method
+    final Map<String, dynamic>  updatesMap = _webSettingsToMap(settings);
+    if (updatesMap.isEmpty) {
```

Comment:
Another option could be `assert(!updatesMap.isEmpty)`

Reply:
It's actually legit to send a `settings` with all null fields to this method.

We could potentially move the short-circuting logic to webview_flutter.dart, but that requires some ugliness or maintaining an `operator==` for `WebSettings` which I'm hesitant to do from a forward-compatibility point of view(I can see us adding a list field to WebSettings, at which point naive looking `webSettings==webSettings2` might actually be expensive.